docs/src/explore/design/pos.md

```diff
@@ -0,0 +1,282 @@
+# Proof of Stake (PoS) system
+
+## Epoch
+
+An epoch is a range of blocks or time that is defined by the base ledger and made available to the PoS system. This document assumes that epochs are identified by consecutive natural numbers. All the data relevant to PoS are [associated with epochs](#epoched-data).
+
+### Epoched data
+
+Epoched data are data associated with a specific epoch that are set in advance. The data relevant to the PoS system in the ledger's state are epoched. Each data can be uniquely identified. These are:
+- [System parameters](#system-parameters). A single value for each epoch.
+- [Active validator set](#active-validator-set). A single value for each epoch.
+- Total voting power. A sum of all active and inactive validators' voting power. A single value for each epoch.
+- [Validators' consensus key, state and total bonded tokens](#validator). Identified by the validator's address.
+- [Bonds](#bonds) are created by self-bonding and delegations. They are identified by the pair of source address and the validator's address.
+
+Changes to the epoched data do not take effect immediately. Instead, changes in epoch `n` are queued to take effect in the epoch `n + pipeline_length` for most cases and `n + unboding_length` for [unboding](#unbond) actions. Should the same validator's data or same bonds (i.e. with the same identity) be updated more than once in the same epoch, the later update overrides the previously queued-up update. For bonds, the token amounts are added up. Once the epoch `n` has ended, the queued-up updates for epoch `n + pipeline_length` are final and the values become immutable.
+
+## Entities
+
+- [Validator](#validator): An account with a public consensus key, which may participate in producing blocks and governance activities. A validator may not also be a delegator.
+- [Delegator](#delegator): An account that delegates some tokens to a validator. A delegator may not also be a validator.
+
+Additionally, any account may submit evidence for [a slashable misbehaviour](#slashing).
+
+### Validator
+
+A validator must have a public consensus key. Additionally, it may also specify optional metadata fields (TBA).
+
+A validator may be in one of the following states:
+- *inactive*:
+  A validator is not being considered for block creation and cannot receive any new delegations.
```

Comment:
but does an inactive validator still vote in consensus (presumably not)?

Reply:
No, they cannot. I'm thinking it would probably be a good idea to rename this state (maybe simply "off"?) to avoid confusion with they validator set, in which we have have active and inactive validator set.

Reply:
yeah, I agree